### DISCUSSION TITLE
increasing timeout doesn't work?

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#21015] [Questions] [answered]
I have a table "chat_room" and then another one "chat_msg" linked to it on its FK room_id (all chat room messages get deleted on cascade when a room is deleted)

Unfortunately when I run the sql "delete from chat_room" it timesout regardless of the "set statement_timeout to xxx;" value I put
there are 500K chat_rooms, 2M messages, so I tried to reduce the query with a :

set statement_timeout to 500000;
DELETE FROM chat_room
WHERE id IN (
    SELECT id
    FROM chat_room
    ORDER BY id
    LIMIT 10000
);

and yet this keeps timing out to, even for just 10K deletion
what am I missing?

EDIT: this is the error:
<html> <head><title>504 Gateway Time-out</title></head> <body> <center><h1>504 Gateway Time-out</h1></center> </body> </html>

**Comment by mwoss** [comment, marked as answer]
Are you running this query from the Supabase SQL editor? If so, the default timeout of 15 sec cannot be changed - see more details in this [SO post here](https://stackoverflow.com/questions/76450610/supabase-sql-editor-upstream-request-timeout). You can workaround this by connecting to DB directly via a client like psql, pgAdmin, TablePlus, or calling this statement via Supabase API.

**Reply by pierroo** [reply]
Huh. Yes I am calling it from the dashboard, is there any reason we cannot increase this limit from there?
As much as possible I try to avoid using too many tools

**Reply by mwoss** [reply]
Good question, I guess this is something that Supabase devs should answer :D

**Reply by pierroo** [reply]
follow up question if you don't mind @mwoss :

I tried to set it up with node js, and once again it times out.
I guess the 8 second default timeout remains unless I increase it, would you know how to increase it from node js?
perhaps there is a parameter to add in createClient? But I cannot find it

EDIT: i'm still curious about the answer, but for the time being "truncate table public.chat_room cascade;" did the trick since I wanted to simply delete everything :)

**Reply by mwoss** [reply]
Have you correctly set a timeout for an API user on the DB level? You can do it via ```alter role authenticator set statement_timeout = 'XXXs';```

**Reply by pierroo** [reply]
I am using the service api key 
in  const supabase = createClient(supa_url, process.env.SUPABASE_SECRET_KEY)

so which role would need to be altered here?

**Reply by mwoss** [reply]
As far as I know the `authenticator`, one from the example above. :D

**Reply by AbdullahAhmadAAK** [reply]
@mwoss I have run `alter role authenticator set statement_timeout = '999s';` in  the SQL editor
But I am still getting this issue. Do you know how this can be resolved?
![image](https://github.com/user-attachments/assets/1b991eb7-6a12-407f-ae4f-cc96a3e89fca)

**Reply by aly-tq** [reply]
It should be 'authenticated' not 'authenticator'!

**Reply by arsenkylyshbek** [reply]
@mwoss @aly-tq this just helped me ship a significant update for our prod feature, thanks a lot!! hahahha
